examples/java/basic_example/src/main/java/Plugin/PluginServerInterceptor.java

```diff
@@ -0,0 +1,69 @@
+package Plugin;
+
+import qlik.sse.ServerSideExtension;
+
+import io.grpc.ServerInterceptor;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.Metadata;
+import io.grpc.ForwardingServerCall.SimpleForwardingServerCall;
+import static io.grpc.Metadata.BINARY_BYTE_MARSHALLER;
+import static io.grpc.Metadata.ASCII_STRING_MARSHALLER;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.Set;
+
+
+public class PluginServerInterceptor implements ServerInterceptor {
+
+    private static final Logger logger = Logger.getLogger(PluginServerInterceptor.class.getName());
+
+    private Plugin.JavaPlugin plugin;
+    
+    public PluginServerInterceptor(Plugin.JavaPlugin plugin) {
+        this.plugin = plugin;
+    }
+
+
+    @Override
+    public <RequestT,ResponseT>ServerCall.Listener<RequestT> interceptCall(
+        ServerCall<RequestT,ResponseT> serverCall, final Metadata metadata, ServerCallHandler<RequestT,ResponseT> serverCallHandler) {
+        logger.finer("Intercepting call to get metadata.");
+        plugin.setMetadata(metadata);
+        logHeader(metadata);
+        return serverCallHandler.startCall(new SimpleForwardingServerCall<RequestT,ResponseT>(serverCall){
+            @Override
+            public void sendHeaders(Metadata responseHeaders) {
+                logger.finest("Send headers.");
+                try {
+                    ServerSideExtension.FunctionRequestHeader header = ServerSideExtension.FunctionRequestHeader
+                    .parseFrom(metadata.get(Metadata.Key.of("qlik-functionrequestheader-bin", BINARY_BYTE_MARSHALLER))); //Java 8
+                    logger.finest("Function request header.");
+                    logHeader(responseHeaders); //Java 8
```

Comment:
What does the "Java 8" comments mean?

Reply:
That you need java 8 for this to work. Should I remove it or make the comment more clear?

Reply:
Prerequisites already states Java 8 as a requirement, so I think it can be removed.